File: backend/src/main/scala/services/review/objects/apiTypes/EligibilityReviewRequest.scala
package services.review.objects.apiTypes

// Business note: protocol DTO shared with the frontend; keep field names and value object types mirrored across stacks.
import services.review.objects.*

import system.objects.given

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*
import system.objects.*

final case class EligibilityReviewRequest(
    target: EligibilityReviewTarget,
    targetId: EntityId,
    reason: ReasonText,
)
object EligibilityReviewRequest:
  given Encoder[EligibilityReviewRequest] = deriveEncoder
  given Decoder[EligibilityReviewRequest] = deriveDecoder

File: frontend/src/objects/review/apiTypes/EligibilityReviewRequest.ts
// Business note: service protocol DTO shared with backend apiTypes; keep field names and value object types aligned.
import type {
  EligibilityReviewTarget,
  EntityId,
  ReasonText,
} from '@/objects/core/SharedObjects'

export type EligibilityReviewRequest = {
  target: EligibilityReviewTarget
  targetId: EntityId
  reason: ReasonText
}
